healthcareai/supervised_model_trainer.py

```diff
@@ -47,6 +47,26 @@ def __init__(self, dataframe, predicted_column, model_type, impute=True, grain_c
         # Split the data into train and test
         self._advanced_trainer.train_test_split()
 
+        # Save the original dataframe columns to the parent class
+        self._advanced_trainer.pre_dummified_columns = dataframe.columns.values
+
+        # Identify the categorical columns
```

Comment:
Good work here. A few thoughts:
1. I wonder if this should be part of the advanced class, since advanced users will likely run into this problem too.
2. This chunk of code from 53 to 66 might do well as its own function, rather than adding to the complexity of the constructor.